tests/drivers/flash_api/testcase.yaml

```diff
@@ -0,0 +1,18 @@
+#
+# Copyright (c) 2024 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+tests:
+  drivers.flash.api:
+    tags: driver flash_api
+    platform_allow:
```

Comment:
just a nitty suggestion, you can ingore: you can move the platform_allow to a common section to save a bit of duplication.